### PR TITLE
Revamped graphics

### PR DESCRIPTION
**Pull request by snouz**
-revamped warehouses and storehouses entities graphics (hr and lr)
-custom icons for each warehouse
-3 new tech icons
-new mod thumbnail
-changed requirement for warehouse-logistics-research-1 from "robotics" to "logistic-robotics" (proposition)
-bots aim for the opening on the warehouse
-wire connector on warehouse appears only when connected
-updated changelog and version

### CODE REVIEW

**Comment by Anoyomouse** [comment]
Looks good so far, I want to do a bit of testing in game, specifically to do with the research, I think you're right needing logistics science for the logistics warehouse

File: VERSION [hunk 1]
```diff
@@ -1 +1 @@
-0.4.1
+0.5
```
Comment (dgw) [inline VERSION:1]: Version bumps are handled outside PRs. (This is also not a valid Factorio mod version number. It would have to be `0.5.0`.)

File: changelog.txt [hunk 2]
```diff
@@ -1,3 +1,14 @@
+---------------------------------------------------------------------------------------------------
+Version: 0.5
+Date: 2021-03-04
+
+  Changed:
+    - New updated graphics by snouz:
+    - New building visuals, HR and LR, with normalized shadows, separate wire connector
+    - New icons for buildings and technologies
+    - New mod icon
+    - Logistic warehouse technology now requires "logistic robotics"
```
Comment (dgw) [inline changelog.txt:10]: This is best summarized in the PR description (top comment). I write the changelog based on what PRs were merged for each release, rather than trying to update the changelog in every PR.

File: prototypes/entity.lua [hunk 3]
```diff
@@ -162,3 +250,9 @@ data:extend({
 })
 
 
+
+
+
+
+
+
```
Comment (dgw) [inline prototypes/entity.lua:258]: We need (or want) exactly 0 of these blank lines. 🙃

File: prototypes/technology.lua [hunk 4]
```diff
@@ -58,7 +59,7 @@ data:extend(
 				recipe = "storehouse-storage",
 			},
 		},
-		prerequisites = { "warehouse-research", "robotics" },
+		prerequisites = { "warehouse-research", "logistic-robotics" },
```
Comment (dgw) [inline prototypes/technology.lua:62]: Can we split this into its own PR?

File: prototypes/technology.lua [hunk 5]
```diff
@@ -2,14 +2,15 @@
  * Part of the Warehousing mod
  *
  * See License.txt in the project directory for license information.
---]]
+--]]
```
Comment (dgw) [inline prototypes/technology.lua:5]: Trailing space added.

**Review by dgw — COMMENTED** [review]
As we've already discussed on the forums, I love this!

The biggest nitpick I have after playing around a bit in-game is that the robot port's shadow on the logistic buildings seems to appear and disappear with the circuit connector's shadow. I don't think you can get away with having only two `-shadow` PNGs; I think you need `-basic-shadow`, `-logistic-shadow`, and `-connector-shadow` layers for each building type.

You'll find in line-notes a few things I'd rather be left to a separate PR or the release process. I have updated the checklist in #51 with all the tasks that _I think_ this PR takes care of. Happy to have input from you on the remaining stuff, like multiple icon sizes, LUTs, and animated warehouse roofs!

Would you be willing to share the layered source files with us? They clearly don't belong in the repo, but if I never have to pixel-edit another sprite like I did to add the Buffer variants a while back, it'll _still_ be too soon! 😂

And finally, beware #87, which will require white sprites/icons to match the vanilla `linked-chest` entity, since someone requested linked warehouses/storehouses to go with that for use in editor scenarios.

**Comment by Anoyomouse** [comment]
> Would you be willing to share the layered source files with us? They clearly don't belong in the repo, but if I never have to pixel-edit another sprite like I did to add the Buffer variants a while back, it'll _still_ be too soon! 😂

I mean, why not, if he's willing to share the source files, then maybe it'll help keeping them around in the repo if we need to edit the sprites, then we have the source layers to help us? (maybe if they add different types of chests or something)

**Comment by snouz** [comment]
> > Would you be willing to share the layered source files with us? They clearly don't belong in the repo, but if I never have to pixel-edit another sprite like I did to add the Buffer variants a while back, it'll _still_ be too soon! 😂
> 
> I mean, why not, if he's willing to share the source files, then maybe it'll help keeping them around in the repo if we need to edit the sprites, then we have the source layers to help us? (maybe if they add different types of chests or something)

I can, but it's 103MB (including source for icons etc.), so where should I transfer it? I can give it to you directly on google drive or something if you'd like. I trust you with it. It's in PSD (CS6 minimum).

For the shadow thing, I can correct this, I just thought it wasn't so important.

I could add a wire connector to the storehouse if you wish, that's a bit more work. Once this PR is resolved, I'll merge and do these small stuffs.

>You'll find in line-notes a few things I'd rather be left to a separate PR or the release process. I have updated the checklist in #51 with all the tasks that I think this PR takes care of. Happy to have input from you on the remaining stuff, like multiple icon sizes, LUTs, and animated warehouse roofs!
Animated roofs (you mean open close the chest part?) I can do, but that'll require a bit of work (I need to make an animation of just the opening based on existing chests and place it exactly right on top of the screen, and for each one),
LUTs, I have no idea what it implies, but I'm curious. I've read the blog post, but I'm curious about it, as it could also improve BioIndustries. For multiple icon sizes (mipmaps), no idea if it's really necessary. Maybe there's a script out there to make them automatically? In bioindustries, I removed them all, it's easier to work with, and I found no visible difference.

**Comment by snouz** [comment]
Well I'm doing some of these corrections right now, adding the connector image to the storehouse too if possible. I'll push these in this PR.

**Comment by snouz** [comment]
I'm making the opening chest animation, so you'll be able to close all points in #51 except LUT (as I said, I don't know what it is)

**Comment by snouz** [comment]
I was very busy yesterday and today, so it took more time than it should, but here it is, chest animations, and corresponding sounds.
I think it's ready now, unless you still have something !

**Comment by snouz** [comment]
BTW, do you need a white version ?
https://github.com/Warehousing/Warehousing/pull/87
I don't know what the white logistic chest is used for, but it'd be pretty quick to export that too.

**Comment by dgw** [comment]
I merged the linked-chest variants, so let me know if you need assistance with a rebase! Replacing the white sprites added by #87 would be much appreciated, yes!

Vanilla Factorio defines a "linked chest" that's only usable in the map editor. I guess scenario authors use them for making the same items available in multiple locations? Whatever the purpose, someone asked for warehouse versions and it was an easy code change, so I did it.

**Comment by snouz** [comment]
>so let me know if you need assistance with a rebase

Ok I'm stuck, it won't let me use the resolve conflict tool on github on my pull request, only command line, but I use github desktop.
https://github.com/snouz/Warehousing/pull/1
Could you help please?

**Comment by dgw** [comment]
@snouz Rebased and cleaned up the whitespace errors etc. while I was at it.

Knowing that you use GitHub Desktop, I'm not sure the best way for you to start using the revised history… GHD is very limited in terms of what it will let you do. The simplest approach might be deleting your local repo and re-cloning. You can verify what changed in the [force-push summary](https://github.com/Warehousing/Warehousing/compare/24be5e5ad209a8f18c2d084dc11bfe0e7d0f9283..4a922d1201ddf97404fb1bc705b2743016ede438) (also linked above in the event message).

----

>>Would you be willing to share the layered source files with us? They clearly don't belong in the repo, but if I never have to pixel-edit another sprite like I did to add the Buffer variants a while back, it'll still be too soon! 😂
>
>I mean, why not, if he's willing to share the source files, then maybe it'll help keeping them around in the repo if we need to edit the sprites, then we have the source layers to help us? (maybe if they add different types of chests or something)

@Anoyomouse It already takes way too long to clone this repo because of the source-assets I added before thinking about long-term impact 😅 and that's only ~25MB of map + high-quality screenshot. A 100MB+ PSD would make life very annoying indeed. I'm honestly tempted to move those source-assets into a separate repo and scrub the history here, but that would be disruptive. For now it's just best to avoid adding any _more_ unnecessary large files to this repo. We can keep the final PSD in its own repo, though? Maybe alongside the tools needed to generate some stuff like the multi-size icons.

**Comment by dgw** [comment]
@snouz Welp, I had to force-push two more times because I missed some whitespace errors… Definitely done now though!

**Comment by snouz** [comment]
Thanks, I think it's now all good !

File: prototypes/entity.lua [hunk 6]
```diff
@@ -96,40 +125,172 @@ data:extend({
...
+	local chestanim = ENTITYPATH..name.."/"..name.."-chest-anim.png"
+	local hrchestanim = ENTITYPATH..name.."/hr-"..name.."-chest-anim.png"
 	p.minable.result = p.name
-	p.icon = "__Warehousing__/graphics/icons/"..p.name..".png"
-	p.picture.filename = "__Warehousing__/graphics/entity/"..p.name..".png"
+	p.icon = ICONPATH..p.name..".png"
+	--p.picture.layers[1].filename = img
+	--p.picture.layers[1].hr_version.filename = hrimg
+	--p.picture.layers[2].filename = shadow
+	--p.picture.layers[2].hr_version.filename = hrshadow
```
Comment (dgw) [inline prototypes/entity.lua:281]: Debug comments? If I'm not lazy I'll remove them, but probably only if I have to go tweak something else.

**Review by dgw — COMMENTED** [review]
I don't expect to have time for the kind of testing I want to give these new sprites until next weekend. But I haven't forgotten about this PR! Just ramping up a big work project for the next week or two.

Note to self below, that I will probably just ignore later. 🙃

File: prototypes/entity.lua (hunk 6 above)
Comment (snouz) [reply prototypes/entity.lua:281]: A remnant from a previous version, until I created an animation instead (to have opening chests)

**Comment by snouz** [comment]
> I don't expect to have time for the kind of testing I want to give these new sprites until next weekend. But I haven't forgotten about this PR! Just ramping up a big work project for the next week or two.

No problem!
I think you should name this release 0.5, as the graphics changes are pretty major for this mod. But that's up to you.
After this pull request is validated, I'll make another one that's just a suggestion: making the second research need logistic-robotics instead of robotics, and making it 120x red&green&blue, instead of 150x red&green, because logistic-robotics uses blue science, and bigger logistic chests should be unlocked after normal logistic chests. What do you think?

**Comment by snouz** [comment]
Have you had the chance to review it? The update shouldn't break anything.

**Comment by dgw** [comment]
> Have you had the chance to review it? The update shouldn't break anything.

It's not breakage that I'd worry about; the quick tests I've done already showed that things still work as expected. What I want to test more is just the experience of playing with the new graphics, to see if there are any obvious tweaks left to make while we still have your attention! My weekend was unexpectedly busy, and left me very little time for anything on the computer (including "real" work), but I promise that I'll take the first chance I have to finish putting these through their paces.

I did have a sec to clean up the history just now though. If I don't find anything to tweak, I'll just merge as soon as I'm done playing around with it. 😸

**Comment by snouz** [comment]
| to see if there are any obvious tweaks left to make while we still have your attention!

I'm not going anywhere : ), I can do modifications after, I'm still contactable everywhere and 2 updates means a lot more downloads, so the mod goes up in the list!

**Comment by dgw** [comment]
Finally had time to play around! This was ready right before a trip, and I had _no_ idea how little free time I'd have while traveling. Returned home earlier this week.

I realized that no one updated the mod's Readme with credit for the revamp, so that's now done. Will merge.

**Comment by dgw** [comment]
@snouz @Anoyomouse Let's figure out where to keep the original working files, in case we need to make tweaks!

**Comment by dgw** [comment]
@snouz Oh, while I'm thinking about it, would it make sense at all to split the graphics up into (more) layers and use the game's ability to tint sprites? I'm wondering if that would save enough file size to be worthwhile, or even just separating the common pieces of the logistic variants from the uniquely colored parts. Future enhancement?

**Comment by JDOGG88** [comment]
@snouz would you be willing to make me a thumbnail for my warehouse compression plugin so it looks basically the same as the base mod? I was thinking just changing the word "MOD" to "COMPRESSION"

**Comment by snouz** [comment]
> @snouz would you be willing to make me a thumbnail for my warehouse compression plugin so it looks basically the same as the base mod? I was thinking just changing the word "MOD" to "COMPRESSION"

Sure !
Here's a fully customized thumbnail, tell me if it works for you
https://i.imgur.com/LyVeVXg.png

**Comment by JDOGG88** [comment]
Perfect! Thanks man :D
